### PR TITLE
ISSUE #3125 - Embedded viewer in v5

### LINKED ISSUE #3125: Embed a functional viewer in v5
### Description
We are facelifting v4 viewer to match the style of v5, as a first step is necessary to have a viewer actually working in v5.

### Goals
- [x] As a user I want to navigate to the viewer in v5
- [ ] As a user I want to be able to do everything that is already working in v4

### Tasks
- [ ] Add basic route for v5 embeded-viewer 
- [ ] Add issue routtes for v5 embeded-viewer 
- [ ] Make sure every functionality work
- [ ] Provide a way to customise the components
- [ ] Make revisions switching work

### PR DESCRIPTION
**Pull request by sanmont3drepo**
This fixes #3125 

#### Description
- Now the viewer has a v5 route, and the viewer is embeded in the v5 layout 
- Routing of issues/risks works in v5
- All of viewer functionality from v4 should work
- Even though the navigation tab is showing on top, the breadcrumbs are not showing the correct data (is missing the model). This will be fixed in a separate issue.

### CODE REVIEW

**Comment by george-hadfield** [comment]
### Initial changes to v5 viewer

image

**Circles**

Changes
image

- [ ] Make default background of circles white
- [ ] Change the icon colour to primary.main on hover (background stays white)
- [ ] Change the active state to secondary.main and the icon colour to primary.main
- [ ] Align the circles vertically underneath the '3D' logo
- [ ] Reduce to padding/margin between the circles to match the design
- [ ] Move the cards a touch closer to the circles (I can't do this in css)

**Toolbar**

Changes
image

- [ ] Change background colour to secondary.main
- [ ] Icons should be positioned horizontally centre inside the toolbar
- [ ] Change the border-radius of the toolbar from 30px to 5px
- [ ] Reduce spacing between icons slightly (I couldn't do this in css)
- [ ] Add a 90% transparency to the background of the toolbar so that model is very slightly visible underneath
- [ ] Ensure the spacing between the bottom of the browser and the bottom of the toolbar is similar to my screenshot at the top of these comments (around 20px)

**Revision select field**

Changes
image

- [ ] Move the revision select field to the main breadcrumb 
- [ ] Should follow the new style breadcrumb

**Comment by sanmont3drepo** [comment]
> ### Initial changes to v5 viewer
> image
> 
> **Circles**
> 
> Changes image
> 
> * [ ]  Make default background of circles white
> * [ ]  Change the icon colour to primary.main on hover (background stays white)
> * [ ]  Change the active state to secondary.main and the icon colour to primary.main
> * [ ]  Align the circles vertically underneath the '3D' logo
> * [ ]  Reduce to padding/margin between the circles to match the design
> * [ ]  Move the cards a touch closer to the circles (I can't do this in css)
> 
> **Toolbar**
> 
> Changes image
> 
> * [ ]  Change background colour to secondary.main
> * [ ]  Icons should be positioned horizontally centre inside the toolbar
> * [ ]  Change the border-radius of the toolbar from 30px to 5px
> * [ ]  Reduce spacing between icons slightly (I couldn't do this in css)
> * [ ]  Add a 90% transparency to the background of the toolbar so that model is very slightly visible underneath
> * [ ]  Ensure the spacing between the bottom of the browser and the bottom of the toolbar is similar to my screenshot at the top of these comments (around 20px)
> 
> **Revision select field**
> 
> Changes image
> 
> * [ ]  Move the revision select field to the main breadcrumb
> * [ ]  Should follow the new style breadcrumb

These comments were moved to a new issue, once this pr is merged in we can start working on these
https://github.com/3drepo/3drepo.io/issues/3161

File: frontend/src/v5/ui/components/dashboard/dashboardList/dashboardList.styles.ts [hunk 1]
```diff
@@ -72,3 +73,13 @@ export const DashedButtonContainer = styled(DashedContainer).attrs(({ theme }) =
...
+
+export const DashboardLink = styled(Link)`
+	&:link { text-decoration: none; }
+	&:visited {
+		text-decoration: none;
+		color:inherit;
+	}
+	&:hover { text-decoration: none; }
+	&:active { text-decoration: none; }
+`;
```
Comment (Amantini1997) [inline frontend/src/v5/ui/components/dashboard/dashboardList/dashboardList.styles.ts:85]: I often happen to apply these CSS styles to `Link`s too, would it be worth it to create a `@control` component that already comes with these styles?

File: frontend/src/v5/ui/themes/global.ts [hunk 2]
```diff
@@ -25,16 +26,40 @@ export const GlobalStyle = createGlobalStyle`
...
+		background: ${COLOR.TRANSPARENT} !important;
+	}
+
+	.emscripten {
+		background: ${COLOR.TRANSPARENT} !important;
+	}
+
+	div#unityViewer.emscripten canvas {
+		background: ${COLOR.TRANSPARENT} !important;
+	}
```
Comment (Amantini1997) [inline frontend/src/v5/ui/themes/global.ts:57]: this and the selector above apply the same rules, would it be possible to join them?`
	.emscripten,
	div#unityViewer.emscripten canvas {
		background: ${COLOR.TRANSPARENT} !important;
	}
`

**Comment by Amantini1997** [comment]
Federations and Containers title look weird
image

**Comment by Amantini1997** [comment]
Federations and Container titles look weird
image

**Comment by Amantini1997** [comment]
When from the viewer I go back by one in the history (to left arrow in the browser view, right under the tabs), I get brought back to v4 
image

File: frontend/src/v5/ui/components/dashboard/dashboardList/dashboardList.styles.ts (hunk 1 above)
Comment (Amantini1997) [reply frontend/src/v5/ui/components/dashboard/dashboardList/dashboardList.styles.ts:85]: Also, adding `color: inherit;` to `&:link` should solve the next mentioned problem

**Comment by Amantini1997** [comment]
The touch-area of a few buttons (in the horizontal bar) is reduced at the bottom. 
image
In other words, all the buttons with the small dash underneath cannot be clicked at the bottom like the other buttons (nor the mouse turns into a cursor). 

To give an idea, the red perimeter delimits the touchable area of the first two buttons
image

This problem, however, exists in staging too for the same buttons
image

**Comment by Amantini1997** [comment]
Clicking on `GIS` triggers an error and the whole app crashes

**Comment by sanmont3drepo** [comment]
> The touch-area of a few buttons (in the horizontal bar) is reduced at the bottom. image In other words, all the buttons with the small dash underneath cannot be clicked at the bottom like the other buttons (nor the mouse turns into a cursor).
> 
> To give an idea, the red perimeter delimits the touchable area of the first two buttons image
> 
> This problem, however, exists in staging too for the same buttons image

Lets move 

> The touch-area of a few buttons (in the horizontal bar) is reduced at the bottom. image In other words, all the buttons with the small dash underneath cannot be clicked at the bottom like the other buttons (nor the mouse turns into a cursor).
> 
> To give an idea, the red perimeter delimits the touchable area of the first two buttons image
> 
> This problem, however, exists in staging too for the same buttons image

I moved this one to issue #3161

**Review by Amantini1997 — APPROVED** [review]

**Pull request merged**